provider/clients.go

```diff
@@ -33,44 +36,89 @@ func GetProviderState() map[string]interface{} {
 	return providerState
 }
 
+// GetClientConfig returns the client.Config produced from the
```

Comment:
Very good points. I tend to take these things for granted, but client-go can be very confusing at first.

It's built with multiple layers of abstraction. The first one is the `RESTClient` which is just a basic HTTP(S) client to the K8s API endpoint that also incorporates authentication. The provider only uses this for fetching the OpenAPI spec since that is just a regular JSON file (no client abstraction around it). The `rest.Config` belongs at this layer too and is how we configure the REST client (HTTP endpoint and auth details). Upper level clients also make use of the RESTClient so even when using those, the same configuration structure applies.

Next level up there is the so called unstructured client, which puts a bit more abstraction on top of the REST client and can deal with individual API resources in a generic manner. No distinction between resource types at this level, they are just represented as `map[string]interface{}`. This is what the alpha provider relies on to make API calls when manipulating resources. Its best suited to the generic approach we take to resource handling, without specific code for each resource type.

Another level up from that are the per-resource clients we're used to from the main provider. You're already familiar with this one.

You were right that the doc comment was misleading. I fixed it to correctly name the type as `rest.Config`.